docs/architecture/PROCESS.md

```diff
@@ -0,0 +1,47 @@
+# ADR Creation Process
+
+1. Copy the `adr-template.md` file. Use the following filename pattern: `adr-next_number-title.md`
+1. Create a draft Pull Request if you want to get an early feedback.
+1. Make sure the context and a solution is clear and well documented.
+1. Add an entry to a list in the [README](./README.md) file.
+1. Create a Pull Request to propose a new ADR.
+
+
+## ADR life cycle
+
+ADR creation is an **iterative** process. Instead of trying to solve all decisions in a single ADR pull request, we MUST firstly understand the problem and collect a feedback through a GitHub Issue.
+
+1. Every proposal SHOULD starts with a new GitHub Issue or be a result of existing Issues. The issue should contain just a brief proposal summary.
+
+1. Once the motivation is validated, a GitHub Pull Request (PR) is created with a new document based on the `adr-template.md`.
+
+1. An ADR doesn't have to arrive to `master` with an _accepted_ status in a single PR. If the motivation is clear and the solution is sound, we SHOULD be able to merge it and keep a _proposed_ status. It's preferable to have an iterative approach rather than long, not merged Pull Requests.
+
+1. If a _proposed_ ADR is merged, then it should clearly document outstanding issues either in ADR document notes or in a GitHub Issue.
+
+1. The PR SHOULD always be merged. In the case of a faulty ADR, we still want to  merge it with a _rejected_ status. The only time the ADR SHOULD NOT be merged is if the author abandons it.
+
+
+### ADR status
+
+```
+DRAFT -> PROPOSED -> LAST CALL yyyy-mm-dd -> FINAL | REJECTED -> SUPERSEEDED by ADR-xxx
+                  \        |
+                   \       |
+                    v      v
+                     ABANDONED
+```
+
+
++ `DRAFT`: [optional] an ADR which is work in progress, not being ready for a general review. This is to present an early work and get an early feedback in a Draft Pull Request form.
++ `PROPOSED`: and ADR covering a full solution architecture.
++ `LAST CALL <date for the last call>`: [optional] clear notify that we are close to accept updates
```

Comment:
This one could be explained into more details:
- What happens on the date of the last call?
- Can we still make changes to the ADR after the last call date?

I fear that because of hard-to-predict schedules from different stakeholders of some ADRs, it's difficult to estimate and enforce a deadline. A social LAST_CALL (via biweekly calls & pinging people on PRs/Discord) has been working quite well imo, and I believe we wouldn't lose much by removing this step, and would have an even more simplified process.

Reply:
It's optional. It's to used to inform others (people who don't participate or miss the calls) that the social consensus is already there, but we still give it a time to digest.

Reply:
I will add more details.